### PR TITLE
Fixed #28041 -- Added prefix matching for PostgreSQL full text search.

### PR DESCRIPTION
**Pull request by joetsoi**
Hi,

I took a stab at putting together something for https://code.djangoproject.com/ticket/28041
I do however, think it is horrendous and I'm not a fan of the way I'm combining `Lexeme`s

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
@joetsoi  what about unit tests?

**Comment by joetsoi** [comment]
Sure, I have some locally that I need to tidy up.

File: django/contrib/postgres/search.py [hunk 1]
```diff
@@ -215,3 +217,121 @@ class TrigramSimilarity(TrigramBase):
...
+
+    def as_sql(self, compiler, connection):
+        param = self.value
+        template = '%s'
+
+        label = ''
+        if self.prefix:
+            label += '*'
+        if self.weight:
+            label += weight
```
Comment (kaedroho) [inline django/contrib/postgres/search.py]: Probably should be ``self.weight``

File: django/contrib/postgres/search.py [hunk 2]
```diff
@@ -215,3 +217,121 @@ class TrigramSimilarity(TrigramBase):
...
+            else:
+                resolved.config = self.config.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        return resolved
+
+    def as_sql(self, compiler, connection):
+        sql, params, = compiler.compile(self.expressions, connection)
+        if self.config:
+            config_sql, config_params = compiler.compile(self.config)
+            template = 'to_tsquery({}::regconfig, {})'.format(config_sql, sql)
+            params = config_params + [self.expressions]
```
Comment (kaedroho) [inline django/contrib/postgres/search.py]: This should probably be: ``params = config_params + params``

**Comment by kaedroho** [comment]
This probably also should escape the Lexemes (removing ``&``, ``|``, etc) before building a query as the following builds an invalid query:
```
RawSearchQuery(Lexeme("A&B") + Lexeme("foo"))
```

**Comment by joetsoi** [comment]
Thanks for looking at this, I'll take a look at fixing this soon

On 22 Aug 2017 2:12 am, "Karl Hobley" <notifications@github.com> wrote:

This probably also should escape the Lexemes (removing &, |, etc) before
building a query as the following builds an invalid query:

RawSearchQuery(Lexeme("A&B") + Lexeme("foo"))

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/django/django/pull/8313#issuecomment-323967209>, or mute
the thread

**Comment by timgraham** [comment]
Hi Joe, do you want to continue this? Test coverage seems lacking and documentation is missing. 
See our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).

**Comment by joetsoi** [comment]
yes, I'll hop on it!

**Comment by LuRsT** [comment]
Any updates @joetsoi?

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**